src/bicep/examples/appServicePlan/appService.bicep

```diff
@@ -0,0 +1,51 @@
+/*
+Deployes a web server farm(aka: App Service Plan) to support web container deployments for Linux.  
+Optionally enable dynamic auto scaling based on CPU Percentages using 'enableAutoScale' true/false
+*/
+targetScope = 'subscription'
+
+@description('The name of the web server farm which will be created.  If unchanged or not specified, the MLZ resource prefix + "--ASP" will be utilized.')
+param appServicePlanName string = 'placeHolder'
+
+@description('The name of the resource group in which the app service plan will be deployed. If unchanged or not specified, the MLZ shared services resource group is used.')
+param targetResourceGroup string = 'placeHolder'
+
+@description('If true, enables dynamic scale-in & scale-out based on CPU percentages.  If false, then compute instances remain static with 2 instances supporting all traffic')
+param enableAutoScale bool = true
+
+var mlzDeploymentVariables = json(loadTextContent('../deploymentVariables.json'))
+var appServicePlanName_Var = contains(appServicePlanName, 'placeHolder') ? '${mlzDeploymentVariables.mlzResourcePrefix.Value}--ASP' : appServicePlanName
+var targetResourceGroup_Var = contains(targetResourceGroup, 'placeHolder') ? '${mlzDeploymentVariables.spokes.Value[2].resourceGroupName}' : targetResourceGroup
+var targetSubscriptionId_Var = contains(targetResourceGroup, 'placeHolder') ? '${mlzDeploymentVariables.spokes.Value[2].subscriptionId}' : subscription().subscriptionId
+var location = deployment().location
+var kind = 'linux'
+var capacity = 2
+var sku = 'premium'
```

Comment:
We can simplify these ternary var assignments on L16-L20 by removing them and allowing for the default parameter override behavior of ARM and Bicep to perform that same evaluation of "Did they specify a parameter value? Okay use that. Did they not specify that parameter? Okay, use the mlzDeploymentVaraibles property instead."

My suggestion below may not be complete, but this is the general idea:

```suggestion
@description('A deploymentVariables.json file that contains MLZ deployment outputs.')
param mlzDeploymentVariables object = json(loadTextContent('../deploymentVariables.json'))

@description('The name of the web server farm which will be created. If unchanged or not specified, the MLZ resource prefix + "-ASP" will be utilized.')
param appServicePlanName string = '${mlzDeploymentVariables.mlzResourcePrefix.Value}-ASP'

@description('The name of the resource group in which the app service plan will be deployed. If unchanged or not specified, the MLZ shared services resource group is used.')
param targetResourceGroupName string = '${mlzDeploymentVariables.spokes.Value[2].resourceGroupName}'

@description('The subscription ID of the resource group in which the app service plan will be deployed. If unchanged or not specified, the MLZ shared services subscription is used.')
param targetSubscriptionId string = '${mlzDeploymentVariables.spokes.Value[2].subscriptionId}'

@description('If true, enables dynamic scale-in & scale-out based on CPU percentages. If false, then compute instances remain static with 2 instances supporting all traffic')
param enableAutoScale bool = true

@description('The region to deploy the app service plan resources into. If unchanged or not specified, the deployment location will be used.')
param location string = deployment().location

var kind = 'linux'
var capacity = 2
var sku = 'premium'
```

Reply:
Updated in latest commit, good change.   I did opt to leave the 'targetSubscriptionID_Var' in place with a slight modification based on your feedback.   The reason why is because I can pick up the subscription through az-context, so it felt like I was asking the user to do the work twice if that makes sense.   What do you think?

Reply:
I think that makes sense 👍